File: C/Algorithm/Graph/DFS/DFS.c
#include<stdio.h>
#include<stdlib.h>
//Define the vertices of the graph
#define V 4
//The function checks whether the vertex has been visited or not
int checkVisited(int visited[], int vertex){
    if(visited[vertex] == 1)
        return 1;
    else
        return 0;
}
//Depth First Search Algorithm
void DFS(int graph[V][V], int start_vertex, int visited[]){
    //Mark current peak visited
    visited[start_vertex] = 1; 
    //Print out the visited vertex
    printf("%d -> ", start_vertex); 
    for(int i = 0; i < V; i++){
        //Check if the i-th vertex is a neighbor of start_vertex
        if(graph[start_vertex][i] == 1){ 
            //If vertex i has not been visited, proceed to visit it
            if(checkVisited(visited, i) == 0){ 
                DFS(graph, i, visited);
            }
        }
    }
}
//Main function to test DFS
int main(){
    //Represent the graph as an adjacency matrix
    int graph[V][V] = {{0,1,1,0}, {1,0,0,1}, {1,0,0,1}, {0,1,1,0}};
    //Array to store visited vertices
    int visited[V] = {0}; 
    printf("Starting from vertex 0 -> ");
    DFS(graph, 0, visited);
    return 0;
}

File: JS/Algorithm/Graph/DFS/DFS.js
//Adjacency graph
const graph = {
  'A': ['B', 'C'],
  'B': ['D', 'E'],
  'C': ['F'],
  'D': [],
  'E': ['F'],
  'F': []
};

//Check function DFS
function dfs(graph, start){
    //Create a set to store the traversed vertices
    const visited = new Set(); 
    function traverse(vertex){
        //If the vertex is already checked, return it
        if (visited.has(vertex)) return; 
        //If you haven't browsed yet, add it to set visited
        visited.add(vertex); 
        //Print out the vertex you are browsing
        console.log(vertex); 
        for(const neighbor of graph[vertex]){ 
            //Traverse the adjacent vertices
            //Recursion into adjacent vertice
            traverse(neighbor); 
        }
    }
    //Start from the top start
    traverse(start);
}
//Traverse the graph from vertex 'A'
dfs(graph, 'A');